src/dswx_sar/refine_with_bimodality.py

```diff
@@ -978,107 +1082,188 @@ def fill_gap_water_bimodality_parallel(
     input_dict : dict
         A dictionary containing file paths for landcover, intensity bands,
         binary water mask, and raster dataset representing land areas.
-    debug_mode: bool
-        If True, additional output metrics and
-        images are saved for debugging purposes.
+    lines_per_block: int
+        Number of lines to be used for the block processing
 
     Returns
     -------
     bimodal_ad_binary : numpy.ndarray
         A binary 2D NumPy array indicating the water gaps.
     """
     rows, cols = meta_info['length'], meta_info['width']
+    input_lines_per_block = lines_per_block
+    pol_str = "_".join(pol_list)
+
+    # To minimize memory usage, the bimodality test will be
+    # carried out with the block first, and entire image will be
+    # used for the remained componenets.
+    lines_per_block_set = [input_lines_per_block, rows]
+    data_shape = [rows, cols]
+    pad_shape = (0, 0)
+
+    fill_gap_path_prefix = 'fill_gap_water_temp'
+    remove_bright_water_path_set = []
+
+
+    for block_iter, lines_per_block in enumerate(lines_per_block_set):
+
+        block_params = dswx_sar_util.block_param_generator(
+            lines_per_block,
+            data_shape,
+            pad_shape)
+        removed_bright_water_path = os.path.join(
+            outputdir,
+            f'{fill_gap_path_prefix}_{pol_str}_{block_iter}.tif')
+        remove_bright_water_path_set.append(removed_bright_water_path)
+
+        for block_ind, block_param in enumerate(block_params):
+            logger.info(f'fill_gap_water_bimodality_parallel block #{block_ind} '
+                        f'from {block_param.read_start_line} to  '
+                        f'{block_param.read_start_line + block_param.read_length}')
+            water_mask = dswx_sar_util.get_raster_block(
+                bright_water_path, block_param)
+            out_boundary = dswx_sar_util.get_raster_block(
+                input_dict['no_data'], block_param)
+            water_mask[out_boundary==1] = 0
+
+            # computes the connected components labeled image of boolean image
+            # and also produces a statistics output for each label
+            nb_components_water, output_water, stats_water, _ = \
+                cv2.connectedComponentsWithStats(np.array(water_mask,
+                                                dtype=np.uint8),
+                                                connectivity=8)
```

Comment:
Same as above - less verbose and alignment nit.
```suggestion
                cv2.connectedComponentsWithStats(water_mask.astype(np.uint8),
                                                 connectivity=8)
```